pkg/blobserver/encrypt/encrypt.go

```diff
@@ -296,7 +296,11 @@ func newFromConfig(ld blobserver.Loader, config jsonconfig.Obj) (bs blobserver.S
 		return nil, errors.New("Must specify passphrase or keyFile")
 	}
 	if keyFile != "" {
-		// TODO: check that keyFile's unix permissions aren't too permissive.
+		if fileInfo, err := os.Stat(keyFile); err != nil {
+			return nil, fmt.Errorf("Checking for key file permissions %v: %v", keyFile, err)
+		} else if fileInfo.Mode().Perm() != 0600 {
```

Comment:
I think this might be too specific for Windows. You probably want `Perm() & 0077 != 0` instead?

Reply:
Ah, done!